### PR TITLE
Distributed Event Bus using Azure

### PR DESCRIPTION
**Pull request by gdeswardt**
### Add Azure Service Bus module

Added an ABP module for Azure Service Bus that implements common
configuration, services, and factories that can be reused in any module.

Added extension methods to the ServiceBusAdministrationClient to check
if a topic and subscription exist and if not create it.

Improved the performance of publishing and processing messages by
creating a pool for ServiceBusAdministrationClient, ServiceBusClient,
ServiceBusSender and ServiceBusProcessor.

### Implement Distributed Event Bus

Implement a new instance of the IDistributedEventBus that uses the
AzureServiceBus module.

Added a service that will replace the existing instance in the service
registry for the distributed event bus with the Azure instance.

Added configuration options for the Azure Event Bus.

### Add Documentation

Added documentation on how to configure and setup the Azure Service Bus
as the Distributed Event Bus.

This resolves ticket #4385

### CODE REVIEW

**Comment by hikalkan** [comment]
Thank you for that great contribution. Sorry, somehow we've missed this PR. I will review and test this in the next days.

**Comment by hikalkan** [comment]
@maliming I also assigned to you, can you review it please?

File: framework/src/Volo.Abp.AzureServiceBus/Volo/Abp/AzureServiceBus/AzureServiceBusMessageConsumer.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+                if (!serviceBusProcessor.IsProcessing)
+                {
+                    await serviceBusProcessor.StartProcessingAsync();
+                }
+
+                while (true)
+                {
+                    Thread.Sleep(1000);
+                }
```
Comment (maliming) [inline framework/src/Volo.Abp.AzureServiceBus/Volo/Abp/AzureServiceBus/AzureServiceBusMessageConsumer.cs:72]: Infinite loop?

Comment (minhhungit) [reply framework/src/Volo.Abp.AzureServiceBus/Volo/Abp/AzureServiceBus/AzureServiceBusMessageConsumer.cs:72]: I guess the code will keep consumer run

**Review by maliming — APPROVED** [review]
I tested it, It wroks.

https://github.com/abpframework/abp-samples/pull/97/files

**Comment by hikalkan** [comment]
@maliming we can merge this, but should wait for outbox/inbox implementation first.

**Review by realLiangshiwei — APPROVED** [review]

**Comment by realLiangshiwei** [comment]
@gdeswardt Thanks again.

**Comment by gdeswardt** [comment]
> @gdeswardt Thanks again.

@realLiangshiwei  only glad I could give back something small to the community. We have been using ABP since version 1. Lots of our clients benefit from this great framework, and hope we could contribute back to the community more in the future.

**Pull request merged**